Quaternion/tests/test_all.py

```diff
@@ -16,6 +89,62 @@ def test_from_eq():
     assert np.allclose(q.q[3],  0.94371436)
     assert np.allclose(q.roll0, 30)
     assert np.allclose(q.ra0, 10)
+    assert q.pitch == -q.dec
+    assert q.yaw == q.ra0
+
+    q1 = Quat(equatorial=[10, 90, 30])
+    q2 = Quat(q=q1.q)
+
+
+def test_from_eq_vectorized():
```

Comment:
The unrolled tests below are excellent.  Can you also add the tests like below, that are the most simple way for my brain to be sure that everything works as expected.  You might need to remove the dec=90 values from the input, not sure.  These are at least partially redundant, but not entirely (I think).
```
atol = 1e-12  # Hopefully this works?

# Compare vectorized computations for all possible input/output combos
# with the same for the scalar calculation.
q = Quat(equatorial=equatorial23)
assert np.all(q.q == q23)
assert np.all(q.equatorial == equatorial23)
assert np.all(q.transform == transform23)

q = Quat(q=q23)
assert np.all(q.q == q23)
assert np.allclose(q.equatorial, equatorial23, rtol=0, atol=atol)
assert np.allclose(q.transform, transform23, rtol=0, atol=atol)

q = Quat(transform=transform23)
assert np.allclose(q.q, q23, rtol=0, atol=atol)
assert np.allclose(q.equatorial, equatorial23, rtol=0, atol=atol)
assert np.all(q.transform == transform23)
```

Reply:
This test fails because of the points I chose, which are gimbal locks.

Then I changed the points to some pseudo-random values and it also fails because q.equatorial can return roll=-30 or roll=330 depending on I don't know what. Angles are not supposed to be normalized? If not, I can check that they are the same module 360 or something like that.

Reply:
and this one fails because some rows in q_23 have negative scalar component, which gets flipped in setter:
```
assert np.all(q.q == q23)
```

Reply:
so... I added a helper function to deal with roll values out of the (0, 360) interval. Perhaps this should be fixed in the class.

I am also checking that q gets flipped when needed.

Reply:
I added this test and it is passing.  Look OK @javierggt ?

Reply:
reply to this is below.